feat: fill days of prev and next month in calendar

d-Rickyy-b opened the pull request:
Instead of only using empty spaces, it would be cool to actually show days of the previous/next month. I implemented a method getPrefill & getPostfill and used them in the njk template. Also both prefill & postfill got their own config value (which defaults to true). They return either empty lists if the setting was set to false or return the days of the prev/next month if the setting was set to true.

d-Rickyy-b committed:
feat: fill days of prev and next month in calendar

Instead of only using empty spaces, it would be cool to actually show days of the previous/next month. I implemented a method getPrefill & getPostfill and used them in the njk template. Also both prefill & postfill got their own config value (which defaults to true). They return either empty lists if the setting was set to false or return the days of the prev/next month if the setting was set to true.

PalatinCoder commented on MMM-MonthCalendar.js:
```diff
@@ -44,22 +46,55 @@ Module.register("MMM-MonthCalendar", {
         for (let day = 1; day <= moment().daysInMonth(); day++) {
             days.push(day);
         }
-        Log.info(days);
```
Good catch. I forgot this one 😄

PalatinCoder reviewed (changes requested):
Implementation is pretty much how I would've done it. However, I think the `preFill` and `postFill` config values should be combined into one. What's the point of showing the next month's days but not the previous ones? 😉

d-Rickyy-b commented:
> 
> 
> Implementation is pretty much how I would've done it. However, I think the `preFill` and `postFill` config values should be combined into one. What's the point of showing the next month's days but not the previous ones? 😉

I was pretty convinced at first to have only one, but then I thought about splitting it for more control. But you are right, why would users want to show any (prev/next) month but not the other 😆 - lemme change that real quick.

d-Rickyy-b committed:
fix: merge prefill and postfill into one option

Now we can either show preview for both, the previous and next month, or none at all.

PalatinCoder reviewed (approved):
LGBR 👍
